library/std/src/lib.rs

```diff
@@ -61,7 +61,10 @@ macro_rules! assert {
         // strategy, which is tracked in
         // https://github.com/model-checking/kani/issues/692
         if false {
+            #[cfg(not(feature = "std"))]
```

Comment:
Comment to the issue this is fixing, for future maintainers to know why we did this

Reply:
Done.